models/note.go

```diff
@@ -0,0 +1,52 @@
+package models
+
+import (
+	"encoding/json"
+	"time"
+)
+
+type NoteId int64
+
+type NoteType int
+
+const (
+	MARGINALIA NoteType = iota
+	META
+	QUESTIONS
+	PREDICTIONS
+)
+
+func (noteType NoteType) String() string {
+	types := [...]string{
```

Comment:
nit: this should probably be outside the function, as an internal package variable, so it doesn't get initialized every time you call the function.